### PR TITLE
feat: use ruff formatter

### PR DESCRIPTION
**Pull request by JacobCoffee**
### Pull Request Checklist

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

### Description
[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

- Testing the swap to using Ruff Formatter

Relevant topics:
- https://github.com/astral-sh/ruff/discussions/7310
- https://docs.astral.sh/ruff/settings/#format
- https://github.com/astral-sh/ruff/issues/1904#issuecomment-1731862483

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

-

### CODE REVIEW

File: tests/unit/test_handlers/test_websocket_handlers/test_handle_websocket.py [hunk 1]
```diff
@@ -28,7 +28,12 @@ class MyController(Controller):
 
         @websocket(path="/", signature_namespace={"d": List[str]})
         async def simple_websocket_handler(
-            self, socket: WebSocket, a: "a", b: "b", c: "c", d: "d"  # type:ignore[name-defined]  # noqa: F821
+            self,
+            socket: WebSocket,
+            a: "a",  # type:ignore[name-defined]  # noqa: F821
+            b: "b",  # type:ignore[name-defined]  # noqa: F821
+            c: "c",  # type:ignore[name-defined]  # noqa: F821
+            d: "d",  # type:ignore[name-defined]  # noqa: F821
```
Comment (JacobCoffee) [inline tests/unit/test_handlers/test_websocket_handlers/test_handle_websocket.py:36]: i hate these

File: tests/unit/test_handlers/test_http_handlers/test_signature_namespace.py [hunk 2]
```diff
@@ -18,7 +18,11 @@ class MyController(Controller):
 
         @decorator(path="/", signature_namespace={"d": List[str], "dict": Dict}, status_code=200)  # type:ignore[misc]
         async def simple_handler(
-            self, a: a, b: b, c: c, d: d  # type:ignore[name-defined]  # noqa: F821
+            self,
+            a: a,  # type:ignore[name-defined]  # noqa: F821
+            b: b,  # type:ignore[name-defined]  # noqa: F821
+            c: c,  # type:ignore[name-defined]  # noqa: F821
+            d: d,  # type:ignore[name-defined]  # noqa: F821
```
Comment (JacobCoffee) [inline tests/unit/test_handlers/test_http_handlers/test_signature_namespace.py:25]: same

File: tests/unit/test_handlers/test_asgi_handlers/test_handle_asgi.py [hunk 3]
```diff
@@ -40,7 +40,10 @@ class MyController(Controller):
 
         @asgi(signature_namespace={"c": Send})
         async def root_asgi_handler(
-            self, scope: "a", receive: "b", send: "c"  # type:ignore[name-defined]  # noqa: F821
+            self,
+            scope: "a",  # type:ignore[name-defined]  # noqa: F821
+            receive: "b",  # type:ignore[name-defined]  # noqa: F821
+            send: "c",  # type:ignore[name-defined]  # noqa: F821
```
Comment (JacobCoffee) [inline tests/unit/test_handlers/test_asgi_handlers/test_handle_asgi.py:46]: same

File: litestar/handlers/base.py [hunk 4]
```diff
@@ -148,9 +148,7 @@ def __init__(
         self.type_encoders = type_encoders
 
         self.paths = (
-            {normalize_path(p) for p in path}
-            if path and isinstance(path, list)
-            else {normalize_path(path or "/")}  # type: ignore
+            {normalize_path(p) for p in path} if path and isinstance(path, list) else {normalize_path(path or "/")}  # type: ignore
```
Comment (peterschutt) [inline litestar/handlers/base.py:151]: I make this 131 chars - maybe it won't break lines if only for a comment?

File: litestar/handlers/http_handlers/_utils.py [hunk 5]
```diff
@@ -147,7 +147,10 @@ def create_response_handler(
     cookie_list = list(cookies)
 
     async def handler(
-        data: Response, app: Litestar, request: Request, **kwargs: Any  # kwargs is for return dto
+        data: Response,
+        app: Litestar,
+        request: Request,
+        **kwargs: Any,  # kwargs is for return dto
```
Comment (peterschutt) [inline litestar/handlers/http_handlers/_utils.py:153]: But then its broken this one which was only 93 chars

**Review by peterschutt — APPROVED** [review]
This looks fine to me. Some hard to follow patterns with breaking statements over multiple lines, but no deal breakers IMO.

File: Makefile [hunk 6]
```diff
@@ -80,6 +80,12 @@ mypy:                                               ## Run mypy
 	@$(ENV_PREFIX)dmypy run
 	@echo "=> mypy complete"
 
+.PHONY: mypy-nocache
+mypy-nocache:                                       ## Run Mypy without cache
+	@echo "=> Running mypy without a cache"
+	@$(ENV_PREFIX)dmypy run -- --cache-dir=/dev/null
```
[the remark this answers is not in the document]
Comment (JacobCoffee) [reply Makefile:86]: the daemon completely restarts because the config is changed

File: litestar/_asgi/routing_trie/mapping.py [hunk 7]
```diff
@@ -191,7 +191,8 @@ def build_route_middleware_stack(
 
     # we wrap the route.handle method in the ExceptionHandlerMiddleware
     asgi_handler = wrap_in_exception_handler(
-        app=route.handle, exception_handlers=route_handler.resolve_exception_handlers()  # type: ignore[arg-type]
+        app=route.handle,  # type: ignore[arg-type]
+        exception_handlers=route_handler.resolve_exception_handlers(),
```
[the remark this answers is not in the document]
Comment (JacobCoffee) [reply litestar/_asgi/routing_trie/mapping.py:195]: I think it is the last trailing comma if it is within the line length

File: litestar/contrib/pydantic/pydantic_schema_plugin.py [hunk 8]
```diff
@@ -159,9 +159,7 @@ def to_openapi_schema(self, field_definition: FieldDefinition, schema_creator: S
         return PYDANTIC_TYPE_MAP[field_definition.annotation]  # pragma: no cover
 
     @classmethod
-    def for_pydantic_model(
-        cls, field_definition: FieldDefinition, schema_creator: SchemaCreator
-    ) -> Schema:  # pyright: ignore
+    def for_pydantic_model(cls, field_definition: FieldDefinition, schema_creator: SchemaCreator) -> Schema:  # pyright: ignore
```
[the remark this answers is not in the document]
Comment (JacobCoffee) [reply litestar/contrib/pydantic/pydantic_schema_plugin.py:162]: I think if the `noqa` or `type: ignore` starts within a certain "inside 120 chars" limit, it will maintain the line

File: litestar/handlers/base.py (hunk 4 above)
Comment (JacobCoffee) [reply litestar/handlers/base.py:151]: same as above, i think you are right. I fought it but ended up confused and commited+pushed

File: Makefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (peterschutt) [reply Makefile:86]: got ya

[the remark this answers is not in the document]
Comment (JacobCoffee) [reply Makefile:86]: Anyone else? @litestar-org/members

File: README.md [hunk 9]
```diff
@@ -11,13 +11,13 @@
...
...roject](https://img.shields.io/badge/Litestar%20Org-%E2%AD%90%20Litestar-202235.svg?logo=python&labelColor=202235&color=edb641&logoColor=edb641)](https://github.com/litestar-org/litestar) [![types - Mypy](https://img.shields.io/badge/types-Mypy-202235.svg?logo=python&labelColor=202235&color=edb641&logoColor=edb641)](https://github.com/python/mypy) [![License - MIT](https://img.shields.io/badge/license-MIT-202235.svg?logo=python&labelColor=202235&color=edb641&logoColor=edb641)](https://spdx.org/licenses/) [![Litestar Sponsors](https://img.shields.io/badge/Sponsor-%E2%9D%A4-%23edb641.svg?&logo=github&logoColor=edb641&labelColor=202235)](https://github.com/sponsors/litestar-org) [![linting - Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/charliermarsh/ruff/main/assets/badge/v2.json&labelColor=202235)](https://github.com/astral-sh/ruff) [![code style - Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/format.json)](https://github.com/psf/black) [![All Contributors](https://img.shields.io/github/all-contributors/litestar-org/litestar?labelColor=202235&color=edb641&logoColor=edb641)](#contributors-) |
```
Comment (JacobCoffee) [inline README.md:20]: code style link is wrong
